meta: clarify nomination process according to Node.js charter (#50834)
nodejs/node: GOVERNANCE.md

Mention @nodejs/collaborators in the issue to notify other collaborators about
the nomination.

> The nomination passes if no collaborators oppose it after one week. In the case of an
> objection, the TSC is responsible for working with the individuals involved and finding
> a resolution.

Reviewer: To fix the linter error
```suggestion
The nomination passes if no collaborators oppose it after one week. In the case
of an objection, the TSC is responsible for working with the individuals
involved and finding a resolution.
```

> involved and finding a resolution.

Reviewer 2: ```suggestion
involved and finding a resolution. The TSC may, at its discretion, vote to override
any objection but should do so only after other measures to reach consensus have
failed.
```

Reviewer 3: I prefer the existing phrasing since it says the TSC is responsible for resolving the matter rather than just voting. The TSC for example can come to consensus about overriding a nomination without voting. It also emphasizes the TSC working with the individuals rather than them having to reach consensus on their own.

> The nomination passes if no collaborators oppose it after one week. Otherwise,
> the nomination fails.
> Maintaining the list of collaborators is a key reponsibility
> of the [Node.js TSC](https://github.com/nodejs/TSC/blob/main/TSC-Charter.md#section-4-responsibilities-of-the-tsc),
> and as a result the TSC can decide to override the opposition under exceptional circumstances, through consensus or voting.

Reviewer 4: ```suggestion
The nomination passes if no collaborators oppose it after one week. In the case of an
objection, the TSC is responsible for working with the individuals involved and finding
a resolution.
```

Reviewer 4: Made a suggestion which uses wording that was similar to what it was before PRs to simply the text.
